contracts/token/ERC721/ERC721.sol

```diff
@@ -192,10 +192,7 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * @dev Returns whether `spender` is allowed to manage `owner`'s tokens, or `tokenId` in
      * particular (ignoring whether it is owned by `owner`).
      *
-     * WARNING: This function doesn't check that `owner` is the actual owner of the specified
-     * `tokenId`. Moreover, it returns true if `owner == spender` in order to skip the check where
-     * needed. Consider checking for cases where `spender == address(0)` since they could lead to
-     * unexpected behavior.
```

Comment:
Why did we simplified this writing? I think it's still important as someone may think by providing a correct `owner` the `spender` doesn't matter. It make sense to me advising for not defaulting to `spender` to `owner`